catroid/src/main/java/org/catrobat/catroid/content/bricks/StitchBrick.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.bricks;
+
+import android.content.Context;
+import android.view.View;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.actions.ScriptSequenceAction;
+
+import java.util.List;
+
+public class StitchBrick extends BrickBaseType {
+
+	private static final long serialVersionUID = 1L;
+
+	public StitchBrick() {
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		return super.getPrototypeView(context);
+	}
+
+	@Override
+	public Brick clone() {
+		return new StitchBrick();
+	}
+
+	@Override
+	public int getViewResource() {
+		return R.layout.brick_stitch;
+	}
+
+	@Override
+	public View getView(Context context) {
+		super.getView(context);
+		return view;
+	}
+
+	@Override
+	public List<ScriptSequenceAction> addActionToSequence(Sprite sprite, ScriptSequenceAction sequence) {
+		sequence.addAction(sprite.getActionFactory().createStitchAction(sprite));
```

Comment:
you could think about changing the ```createStitchAction(sprite)``` to a static method in the ```ActionFactory```, there is no internal state anyway so you don't really need an instance of the ```ActionFactory``` here.

Reply:
I thought you were going to protest against this proposal :thinking:  because now that I had another look at it, it would be inconsistent. anyway, in this case either way is fine with me, so you could also change it back if you want. :roll_eyes: